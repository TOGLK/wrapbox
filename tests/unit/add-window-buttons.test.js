const addWinButtons = require("../../app/scripts/add-window-buttons");
const { ipcRenderer } = require("electron"); // eslint-disable-line

jest.mock("electron", () => require("../mocks/electron"));

const titleBar = document.createElement("div");
titleBar.setAttribute("id", "titlebar");
document.querySelector("body").appendChild(titleBar);

addWinButtons("#titlebar", "right");
const winButtons = document.querySelector("#close").parentNode;

describe("AddWinButtons function", () => {
  test("adds window buttons to titlebar", () => {
    expect(winButtons.parentNode.id).toBe("titlebar");
  });
  test("adds 3 buttons", () => {
    const buttons = winButtons.childNodes;
    expect(buttons).toHaveLength(3);
    for (let i = 0; i < buttons.length; i++) {
      expect(buttons[i]).toMatchSnapshot();
    }
  });

  describe("Window buttons", () => {
    test("parent has class 'right'", () => {
      expect(winButtons.classList.contains("right")).toBeTruthy();
    });
    test("send messages through ipcRenderer", () => {
      winButtons.childNodes.forEach((e) => {
        e.click();
      });
      expect(ipcRenderer.send.mock.calls).toEqual([["minimize"], ["maximize"], ["close"]]);
    });
  });
});
