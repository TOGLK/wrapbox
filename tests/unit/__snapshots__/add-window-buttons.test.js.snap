// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddWinButtons function adds 3 buttons 1`] = `
<button
  class="mdl-button mdl-js-button mdl-color-text--white"
  id="minimize"
>
  <i
    class="material-icons"
  >
    expand_more
  </i>
</button>
`;

exports[`AddWinButtons function adds 3 buttons 2`] = `
<button
  class="mdl-button mdl-js-button mdl-color-text--white"
  id="maximize"
>
  <i
    class="material-icons"
  >
    expand_less
  </i>
</button>
`;

exports[`AddWinButtons function adds 3 buttons 3`] = `
<button
  class="mdl-button mdl-js-button mdl-color-text--white"
  id="close"
>
  <i
    class="material-icons"
  >
    close
  </i>
</button>
`;
